client/ayon_core/lib/file_transaction.py

```diff
@@ -212,3 +226,46 @@ def _same_paths(self, src, dst):
             return os.stat(src) == os.stat(dst)
 
         return src == dst
+
+
+def as_completed_stop_and_raise_on_error(
```

Comment:
The function name is wierd, it does not tell for what it should be called, maybe something like `wait_for_futures`, without the "raise on error".